types/index.d.ts

```diff
@@ -29,9 +31,11 @@ export type Overlap<T extends string, U extends string> = Diff<T, Diff<T, U>>;
  * type from U only.
  *
  * @see https://github.com/pelotom/type-zoo/issues/2
- * @see https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-307871458
+ * @see https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-377692897
  */
-export type Overwrite<T, U> = Omit<T, Overlap<keyof T, keyof U>> & U;
+export type Overwrite<T, U> = {
```

Comment:
I don't see why this one needs to change except to use `Extract` instead of `Overlap`:

```ts
export type Overwrite<T, U> = Omit<T, Extract<keyof T, keyof U>> & U;
```